ui/src/components/setting/SettingWebhook.vue

```diff
@@ -0,0 +1,189 @@
+          <v-col
+            md="auto"
+            class="ml-auto"
+          >
+            <v-btn
+              v-if="!editWebhook"
+              outlined
+              @click="editWebhook = !editWebhook"
+            >
+              Change webhook
+            </v-btn>
+            <div
+              v-if="editWebhook"
+            >
+              <v-btn
+                class="mr-2"
+                outlined
+                @click="cancel"
+              >
+                Cancel
+              </v-btn>
+              <v-btn
+                outlined
+                @click="passes(save)"
+              >
+                Save
+              </v-btn>
+            </div>
+          </v-col>
+        </v-row>
+        <div
+          class="pl-4 pr-4"
+          v-for="(value, name) in webhookItem"
+          :key="name"
+        >
+          <v-row>
+            <v-col>
+                <ValidationProvider
+                  v-slot="{ errors }"
+                  :ref="`provider-${name}`"
+                  :vid="name"
+                  name="Priority"
+                >
+                  <v-text-field
+                    v-model="webhookItem[name]"
+                    :data-test="`field-${name}`"
+                    :label="name"
+                    :disabled="!editWebhook"
+                    :error-messages="errors"
+                  />
+                </ValidationProvider>
+            </v-col>
+          </v-row>
+        </div>
+        <div class="pl-4 pr-4">
+          <v-row>
+            <div class="ml-3 mb-1">
+              <v-checkbox
+                v-model="enableWebhook"
+                :disabled="!editWebhook"
+                label="Enable webhook"
+              />
+            </div>
+          </v-row>
+          <v-row class="mb-2">
+            <div class="ml-3 mb-6">
+              <p>
+                Set a webhook to receive an alert when you try to connect
+                to a device and start up the agent.
+              </p>
+            </div>
+          </v-row>
+        </div>
+        <v-spacer />
+      </ValidationObserver>
+    </v-form>
+  </fragment>
+</template>
+
+<script>
+
+import {
+  ValidationObserver,
+  ValidationProvider,
+} from 'vee-validate';
+
+export default {
+  name: 'SettingWebhook',
+
+  components: {
+    ValidationObserver,
+    ValidationProvider,
+  },
+
+  data() {
+    return {
+      enableWebhook: false,
+      missingWh: false,
+      webhookItem: {},
+      editWebhook: false,
+    };
+  },
+
+  computed: {
+    tenant() {
+      return localStorage.getItem('tenant');
+    },
+
+    webhook() {
+      return this.$store.getters['namespaces/webhook'];
+    },
+  },
+
+  created() {
+    this.setWebhookData();
+  },
+
+  methods: {
+    async updateWebhook() {
+      try {
+        const {
+          port,
+          url,
+          scheme,
+          uri,
+        } = this.webhookItem;
+
+        await this.$store.dispatch('namespaces/updateWebhook', {
```

Comment:
You can create const to receive data.
const webhook = {
            url,
            port: parseInt(port, 10),
            scheme,
            uri,
            active: this.enableWebhook,
          };
          
          await this.$store.dispatch('namespaces/updateWebhook', { this.tenant, webhook});